glitch/analysis/security.py

```diff
@@ -1,15 +1,1107 @@
+                        name = storage_account_name.value.lower().split('.')[1]
+                        storage_account_au = self.get_au(code, file, name, "resource.azurerm_storage_account")
+                        if storage_account_au:
+                            expr = "\${azurerm_storage_account\." + f"{name}\."
+                            pattern = re.compile(rf"{expr}")
+                            assoc_au = self.get_associated_au(code, file, "resource.azurerm_log_analytics_storage_insights",
+                                "storage_account_id", pattern, [""])
+                            if assoc_au:
+                                blob_container_names = self.check_required_attribute(assoc_au.attributes, [""], "blob_container_names[0]")
+                                if blob_container_names:
+                                    i = 0
+                                    contains_blob_name = False
+                                    while blob_container_names:
+                                        a = blob_container_names
+                                        if blob_container_names.value:
+                                            contains_blob_name = True
+                                            break
+                                        i += 1
+                                        blob_container_names = self.check_required_attribute(assoc_au.attributes, [""], f"blob_container_names[{i}]")
+                                    if not contains_blob_name:
+                                        errors.append(Error('sec_logging', a, file, repr(a)))
+                                else:
+                                    errors.append(Error('sec_logging', assoc_au, file, repr(assoc_au), 
+                                    f"Suggestion: check for a required attribute with name 'blob_container_names'."))
+                            else:
+                                errors.append(Error('sec_logging', storage_account_au, file, repr(storage_account_au), 
+                                    f"Suggestion: check for a required resource 'azurerm_log_analytics_storage_insights' " + 
+                                    f"associated to an 'azurerm_storage_account' resource."))
+                        else:
+                            errors.append(Error('sec_logging', element, file, repr(element), 
+                                f"Suggestion: 'azurerm_storage_container' resource has to be associated to an " + 
+                                f"'azurerm_storage_account' resource in order to enable logging."))
+                    else:
+                        errors.append(Error('sec_logging', element, file, repr(element), 
+                            f"Suggestion: 'azurerm_storage_container' resource has to be associated to an " + 
+                            f"'azurerm_storage_account' resource in order to enable logging."))
```

Comment:
Here is another case that would benefit from a refactor. Using the function for the `while` would help a bit. Other than that, maybe everything inside the `elif` should be in a function. In that way, when one of the ifs failed, the function could return and every block `else` would be removed.